Drafting R Specific Standards

RGilliard-Arch opened the pull request:
Closes #3

R-Gilliard committed:
Add r_standards.md

R-Gilliard committed:
Correct links and guideline intro

janejuenyang commented on r_standards.md:
```diff
@@ -0,0 +1,50 @@
+## Pipes
+
+### Assignment
+- Avoid right-handed assignment
```
FWIW, I prefer using "do" vs. "don't" framing where possible. So in this case it'd be "Use left-handed assignment"

janejuenyang commented on r_standards.md:
```diff
@@ -0,0 +1,50 @@
+## Packages
+
+### Package Management
+We will use [renv](https://rstudio.github.io/renv/articles/renv.html) to manage packages for our projects/code written in R. When beginning a new project, create a dedicated R project. This R project, and attendant renv related files, should be tracked on GitHub so that all contributors are working in the same development environment. See the [renv](https://rstudio.github.io/renv/articles/renv.ht ...
```
In second URL, please include "documentation" into the hyperlink text to make it more accessible.

janejuenyang commented on r_standards.md:
```diff
@@ -0,0 +1,50 @@
+### Preferred Packages
+
+- Data Cleaning: [tidyverse](https://www.tidyverse.org/), [dplyr](https://dplyr.tidyverse.org/), [tidyr](https://tidyr.tidyverse.org/), [arrow](https://arrow.apache.org/docs/r/), [dtplyr](https://dtplyr.tidyverse.org/), [janitor](https://www.rdocumentation.org/packages/janitor/versions/2.2.0)
+    - Installing the tidyverse will install both dplyr and tidyr, as well as many other useful packages. However, we encourage parsimony. That is, if only a few of the packages in the tidyverse are needed, do not load them all. To that end we listed packages in the tidyverse individually throughout this guide.
```
TIL "parsimony" :) Can you add a short reason why (basically performance...)

janejuenyang commented on r_standards.md:
```diff
@@ -0,0 +1,50 @@
+- Loops: [purrr](https://purrr.tidyverse.org/), [furrr](https://furrr.futureverse.org/)
+    - It is acceptable to use the base R `*apply()` suite of functions for looping.
+    - Use furrr as a drop-in replacement for purrr if attempting to parallelize code.
+- RMarkdown: [knitr](https://yihui.org/knitr/), [tinytex](https://yihui.org/tinytex/)
```
What does the team think about [Quarto](https://r4ds.hadley.nz/quarto.html) in lieu of RMarkdown? Hadley Wickham and co call it "clearly the future" in their [preface to _R for Data Science_](https://r4ds.hadley.nz/preface-2e)

janejuenyang commented on r_standards.md:
```diff
@@ -0,0 +1,50 @@
+    - It is acceptable to use Base R where possible, but we recommend considering readr for its speed and consistent naming conventions.
+- Machine Learning/Statistics: [tidymodels](https://www.tidymodels.org/)
+- Geospatial Analyses: [sf](https://r-spatial.github.io/sf/), [terra](https://github.com/rspatial/terra)
+- String Manipulation: [stringr](https://stringr.tidyverse.org/)
```
Please add a section for visualizations. ggplot is the obvious one. Defer to the team if you want to add any other preferred packages for interactive graphics.

janejuenyang reviewed (changes requested):
Great start, thanks @R-Gilliard ! Mostly small comments -- only one that might take a moment to sync with the team on is Quarto vs. RMarkdown.

R-Gilliard committed:
Addressing PR Comments

Include "documentation" in hyperlink text
Additional description for parsimony in tidyverse
Visualizations section
Dashboard section
Swap to Quarto rather than R Markdown
Use positive framing in assignment section

RGilliard-Arch commented:
@janejuenyang I addressed your comments in this commit [12c1618](https://github.com/HHS/acf-datasurge-standards/pull/4/commits/12c1618bcf77f672b65a3a8ae4efe6cbaa0335f9). Happy to wait to merge this for feedback from other folks regarding whether to use Quarto or RMarkdown, but I feel comfortable defaulting to Quarto, particularly because of its higher degree of language agnosticism.

janejuenyang commented on r_standards.md:
```diff
@@ -18,33 +18,38 @@ my_mean <- function(x) {
 
 - Data Cleaning: [tidyverse](https://www.tidyverse.org/), [dplyr](https://dplyr.tidyverse.org/), [tidyr](https://tidyr.tidyverse.org/), [arrow](https://arrow.apache.org/docs/r/), [dtplyr](https://dtplyr.tidyverse.org/), [janitor](https://www.rdocumentation.org/packages/janitor/versions/2.2.0)
-    - Installing the tidyverse will install both dplyr and tidyr, as well as many other useful packages. However, we encourage parsimony. That is, if only a few of the packages in the tidyverse are needed, do not load them all. To that end we listed packages in the tidyverse individually throughout this guide.
+    - Installing the tidyverse will install both dplyr and tidyr, as well as many other useful packages. However, we encourage parsimony. That is, if only a few of the packages in the tidyverse are needed, do not load them all. This helps with reproducibility, making clear what pacakges are needed for a specific script, and with performance. To that end we listed packages in the tidyverse individ ...
```
small typo "pacakges"

janejuenyang commented on r_standards.md:
```diff
@@ -18,33 +18,38 @@ my_mean <- function(x) {
     - It is acceptable to use the base R `*apply()` suite of functions for looping.
     - Use furrr as a drop-in replacement for purrr if attempting to parallelize code.
-- RMarkdown: [knitr](https://yihui.org/knitr/), [tinytex](https://yihui.org/tinytex/)
+- Quarto: [rmarkdown](https://www.rdocumentation.org/packages/rmarkdown/versions/2.25)
```
Suggest also linking to https://quarto.org/

janejuenyang reviewed (approved):
Thanks for the updates, Reggie! Two small things, feel free to merge after addressing those.

R-Gilliard committed:
Add quarto link and correct spelling
